---
title: "Tumors that respond poorly to bevacizumab show upregulation of angiogenesis genes."
author: "Roshan Lodha"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  github_document:
    toc: false
    toc_depth: 4
  #html_document:
  #  df_print: paged
  #  keep_md: true
  #pdf_document: default
keyword: "bevacizumab"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(warning = FALSE, # turn off warnings
                      message = FALSE,
                      echo = FALSE,
                      results = 'hide') # hide console output
knitr::opts_chunk$set(fig.width = 10, fig.height = 7) # set figure height and width
```

# Bulk RNA-sequencing analysis

## Loading required packages and tools

```{r load-packages}
for (package in c('BiocManager', 'tidyverse', 'readxl', 'matrixStats', 'cowplot', 'DT', 'plotly', 'gt', 'ggrepel', 'gprofiler2', 'ggthemes', 'ggprism', 'gtsummary', 'ggforce', 'ggdendro', 'ggpubr', 'umap', 'ashr')) {
  if (!require(package, character.only = T, quietly = T)) {
    install.packages(package,
      repos = "http://cran.us.r-project.org")
    library(package, character.only = T)
  }
}

bio_pkgs <- c("biomaRt", "tximport", "ensembldb", "EnsDb.Hsapiens.v86", "edgeR", "DESeq2", "limma", "apeglm", "GSEABase", "Biobase", "GSVA", "clusterProfiler", "msigdbr", "enrichplot", "annotate", "org.Hs.eg.db", "TCGAbiolinks")
surv_pkgs <- c("UCSCXenaTools", "dplyr", "survival", "survminer", "ggbreak", "ggprism", "svglite", "tidyverse")
#BiocManager::install(bio_pkgs)
#BiocManager::install(surv_pkgs)
invisible(lapply(bio_pkgs, function (x) library(x, character.only = T)))
invisible(lapply(surv_pkgs, function (x) suppressMessages(library(x, character.only = T))))
```

```{r load-expr}
design <- read_tsv("input/prefilterstudydesign.txt")
sampleLabels <- design$sample
group <- factor(design$group)

gbmexpr <- read_csv("input/prefiltergbmexpr.csv")[2: 13] #pre-filtered
```

### Specific Expression Analysis
```{r specific-sample-expr}
gbmexpr.select <- gbmexpr %>% 
  dplyr::filter(`Gene Symbol` == "PTEN")
gbmexpr.select
```

## Preprocessing 

```{r preprocessing}
gbmexpr.matrix <- as.matrix(gbmexpr[, -1])
rownames(gbmexpr.matrix) <- unlist(gbmexpr[, 1])
myDGEList <- DGEList(gbmexpr.matrix)
myDGEList.filtered.norm <- calcNormFactors(myDGEList, method = "TMM") #normalize using TMM
log2.tpm.filtered.norm <- log2(gbmexpr.matrix + 1)
log2.tpm.filtered.norm.df <- as_tibble(log2.tpm.filtered.norm, rownames = "geneID")
```

```{r preprocessing-figs, fig.height=4, fig.width=10}
distance <- dist(t(log2.tpm.filtered.norm), method = "maximum")
clusters <- hclust(distance, method = "average")

suppfig1a <- ggdendrogram(clusters) +
  xlab("Distance") +
  ylab("Sample") +
  theme_prism() +
  theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1))

sampleLabels <- substr(sampleLabels, 1, nchar(sampleLabels) - 5)
pca.res <- prcomp(t(log2.tpm.filtered.norm), scale. = F, retx = T)
pc.var <- pca.res$sdev ^ 2
pc.per <- round(pc.var / sum(pc.var) * 100, 1)
pca.res.df <- as_tibble(pca.res$x)
pca.res.df$sampleLabel = sampleLabels
pca.res.df$group = group
suppfig1b <- ggplot(pca.res.df) +
  aes(x = PC1, y = PC2, color = group) +
  geom_point(size = 3) +
  geom_label_repel(aes(label = sampleLabels), hjust = 0.1, vjust = 0.1) +
  scale_color_manual(values = c("#ffb464", "#126079")) +
  xlab(paste0("PC1 (", pc.per[1], "%", ")")) +
  ylab(paste0("PC2 (", pc.per[2], "%", ")")) +
  theme_prism() +
  theme(legend.position = c(0.9, 0.5))
```

```{r suppfig1}
suppfig1 <- ggarrange(suppfig1a, suppfig1b, ncol = 2, nrow = 1)
suppfig1
```

# Unsupervised Analysis

```{r assign-based-on-pc}
# Assign clusters based on PC1 value
pca.res.df$cluster <- ifelse(pca.res.df$PC1 > 0, "Cluster 2", "Cluster 1")

# Convert to a dataframe with sample labels
cluster_df <- data.frame(sample = pca.res.df$sampleLabel, 
                         cluster = factor(pca.res.df$cluster))

# Visualize PCA with new clusters
ggplot(pca.res.df, aes(x = PC1, y = PC2, color = cluster, label = sampleLabel)) +
  geom_point(size = 3) +
  geom_text_repel() +
  scale_color_manual(values = c("Cluster 1" = "#126079", "Cluster 2" = "#ffb464")) +
  ggtitle("Clusters Based on PC1") +
  theme_prism()

```

```{r unsup-assign}
design <- cbind(design, cluster_df %>% select(c("cluster")))
sampleLabels <- design$sample
group <- factor(design$group)
mm <- model.matrix(~0 + cluster, data = design)
``` 

```{r unsup-dge}
counts <- gbmexpr.matrix
counts <- counts[rowSums(counts <= 0) <= 3, ] %>% drop_na() #filter: at most 3 zeros
rownames(counts.matrix) <- counts$gene

dds <- DESeqDataSetFromMatrix(countData = round(counts),
                              colData = design,
                              design = ~ cluster)

keep <- rowSums(counts(dds)) >= 350  
dds <- dds[keep, ]

dds <- DESeq(dds)
res <- results(dds, contrast = c("cluster", "Cluster 2", "Cluster 1"))
res <- lfcShrink(dds, type = "ashr", contrast = c("cluster", "Cluster 2", "Cluster 1"))

deseq_pc1 <- as.data.frame(res) %>%
  drop_na() %>%
  arrange(padj) %>%
  arrange(desc(abs(log2FoldChange)))

deseq_pc1$gene <- getSYMBOL(rownames(deseq_pc1), data = 'org.Hs.eg')
deseq_pc1 <- dplyr::filter(deseq_pc1, gene %in% coding_genes)

deseq_pc1 <- deseq_pc1 %>%
  mutate(enrichment = case_when(
    (padj < 0.05) & (log2FoldChange > 1) ~ "Cluster 2",
    (padj < 0.05) & (log2FoldChange < -1) ~ "Cluster 1"
  ))
deseq_pc1$enrichment[is.na(deseq_pc1$enrichment)] <- "none"
rownames(deseq_pc1) <- deseq_pc1$gene
```

## Postprocessing

```{r redefine-counts-matrix}
design <- read_tsv("./input/studydesign.txt")
sampleLabels <- design$sample
group <- factor(design$group)
mm <- model.matrix(~0 + group)

counts <- read.table(file = "./input/counts.tabular", header = TRUE, sep = "\t")
colnames(counts) <- c("geneID", sampleLabels)
counts$gene <- getSYMBOL(as.character(counts$geneID), data = 'org.Hs.eg')
counts <- counts %>% select(gene, everything())
counts <- counts[rowSums(counts <= 0) <= 3, ] %>% drop_na() #filter: at most 3 zeros
counts.matrix <- as.matrix(counts[3: 11])
rownames(counts.matrix) <- counts$gene
```

```{r pca}
pca.res <- prcomp(t(log(counts.matrix + 1)), scale. = F, retx = T)
pc.var <- pca.res$sdev ^ 2
pc.per <- round(pc.var / sum(pc.var) * 100, 1)
pca.res.df <- as_tibble(pca.res$x)
pca.plot <- ggplot(pca.res.df) +
  aes(x = PC1, y = PC2, label = sampleLabels, color = group) +
  geom_point(size = 3) +
  geom_text(aes(label = sampleLabels), hjust = 0, vjust = 0) +
  scale_color_manual(values = c("#ffb464", "#126079")) +
  xlab(paste0("PC1 (", pc.per[1], "%", ")")) +
  ylab(paste0("PC2 (", pc.per[2], "%", ")")) +
  stat_ellipse() +
  theme_prism()
```

## Bulk-RNA Analysis

### DGE Plot
```{r suppfig2}
#mart = useEnsembl(biomart='ensembl', dataset = "hsapiens_gene_ensembl", mirror = "useast")
mart <- useMart(biomart = "ensembl", 
         host = "www.ensembl.org")
mart <- useMart("ENSEMBL_MART_ENSEMBL") #coding genes for cleaning
mart <- useDataset("hsapiens_gene_ensembl", mart)

coding_genes <- getBM(attributes = c("hgnc_symbol"),
  filters = c("biotype"),
  values = list(biotype = "protein_coding"),
  mart = mart)$hgnc_symbol

rownames(counts.matrix) <- counts$geneID
dds <- DESeqDataSetFromMatrix(countData = round(counts.matrix),
  colData = design,
  design = ~group)

keep <- rowSums(counts(dds)) >= 350 #determined via hyperparameter exploration
dds <- dds[keep, ]
dds <- DESeq(dds)
dds <- estimateSizeFactors(dds)
deseqvoom <- voom(counts(dds, normalized = TRUE), mm, plot = T)
```

```{r dge, results='hold', message=FALSE}
res <- results(dds)
res <- lfcShrink(dds, coef = "group_poor_vs_good", type = "ashr")
deseq <- as.data.frame(res) %>% drop_na() %>% arrange(padj) %>% arrange(desc(abs(log2FoldChange)))
deseq$gene <- getSYMBOL(rownames(deseq), data = 'org.Hs.eg')
deseq <- dplyr::filter(deseq, gene %in% coding_genes)
deseq <- deseq %>%
  mutate(enrichment = case_when(
    (padj < 0.05) & (log2FoldChange > 1) ~"poor",
    (padj < 0.05) & (log2FoldChange <- 1) ~"good"))
deseq$enrichment[is.na(deseq$enrichment)] <- "none"
rownames(deseq) <- deseq$gene
```

```{r dge-plot, fig.show='hide'}
fig1a <- ggplot(deseq) +
  aes(y = -log10(padj), x = log2FoldChange, colour = enrichment) +
  scale_color_manual(values = c("good" = "#126079", "none" = "grey", "poor" = "#ffb464")) +
  geom_point(size = 1.5, alpha = 0.25) +
  geom_rect(mapping = aes(xmin = 9, xmax = 11, ymin = 2, ymax = 54), alpha = 0, color = 'black') +
  geom_text_repel(size = 4, data = head(deseq, 20), aes(label = gene), max.overlaps = Inf, colour = "black") +
  geom_hline(yintercept = -log10(0.05), linetype = "longdash", colour = "black", size = .5) +
  geom_text(aes(-8, -log10(0.05), label = "p = 0.05", vjust = 1), colour = "black") +
  xlab("Fold Change (log2)") +
  ylab("Significance (log10)") +
  theme_prism() +
  theme(legend.position = c(0.15, 0.8))
fig1a
```

### GSEA Plot

```{r gsea, results='hold'}
hs_gsea <- msigdbr(species = "Homo sapiens")
hs_gsea %>% dplyr::distinct(gs_cat, gs_subcat) %>% dplyr::arrange(gs_cat, gs_subcat)
hs_gsea_h <- msigdbr(species = "Homo sapiens",
    category = "H") %>%
  dplyr::select(gs_name, gene_symbol)
hs_gsea_kegg <- msigdbr(species = "Homo sapiens",
    category = "C2",
    subcategory = "CP:KEGG") %>%
  dplyr::select(gs_name, gene_symbol)
hs_gsea_tcga_gbm <- msigdbr(species = "Homo sapiens",
    category = "C2",
    subcategory = "CGP") %>%
  dplyr::select(gs_name, gene_symbol)

deseq.GSEA.select <- dplyr::select(deseq, gene, log2FoldChange, padj)
deseq.gsea <- abs(deseq.GSEA.select$log2FoldChange) / deseq.GSEA.select$log2FoldChange * -log10(deseq.GSEA.select$padj)
names(deseq.gsea) <- as.character(deseq.GSEA.select$gene)
deseq.gsea <- sort(deseq.gsea, decreasing = TRUE)
deseq.gsea.res <- GSEA(deseq.gsea, pvalueCutoff = 1, TERM2GENE = hs_gsea_h, verbose = FALSE)
deseq.GSEA.df <- as_tibble(deseq.gsea.res @result)
deseq.GSEA.df <- deseq.GSEA.df %>%
  mutate(phenotype = case_when(
    (NES > 0) & (p.adjust < 0.05) ~"poor",
    (NES < 0) & (p.adjust < 0.05) ~"good"))
deseq.GSEA.df$phenotype[is.na(deseq.GSEA.df$phenotype)] <- "none"
deseq.GSEA.df$Description <- gsub("_", " ", deseq.GSEA.df$Description)
head(deseq.GSEA.df, 25) %>%
  dplyr::select(-c("ID", "enrichmentScore", "pvalue", "qvalue", "rank", "leading_edge")) %>%
  gt()
```

```{r hgsea-plot, fig.show='hide'}
fig1b <- ggplot(deseq.GSEA.df, aes(x = NES, y = -log10(p.adjust), color = phenotype)) +
  geom_point(aes(size = setSize), alpha = 0.5) +
  scale_color_manual(values = c("good" = "#126079", "none" = "grey", "poor" = "#ffb464")) +
  geom_text(aes(-2, -log10(0.05), color="black", label = "p = 0.05", vjust = -1)) +
  geom_hline(yintercept = -log10(0.05), linetype = "longdash", size = .5, color="black") +
  ylab("Significance (log10)") +
  xlab("Normalized Enrichment Score") +
  theme_prism() +
  theme(legend.position = c(0.8, 0.7),
        legend.title = element_text())
fig1b
```

```{r suppfig3}
suppfig3a <- ggplot(deseq.GSEA.df %>% dplyr::filter(phenotype == "good")) +
  geom_col(aes(x = reorder(Description, -NES), y = NES, fill = NES)) +
  coord_flip() + 
  scale_fill_gradient2(low = "red", mid = "blue", high = "blue", midpoint = 0) +
  labs(title = "Downregulated in Poor Responders") +
  xlab("Hallmark Pathways") + 
  ylab("Normalized Enrichment Score (NES)") + 
  guides(fill = FALSE) +
  theme_prism(base_fontface = "plain")

suppfig3b <- ggplot(deseq.GSEA.df %>% dplyr::filter(phenotype == "poor")) +
  geom_col(aes(x = reorder(Description, -NES), y = NES, fill = NES)) +
  coord_flip() + 
  scale_fill_gradient2(low = "red", mid = "blue", high = "blue", midpoint = 0) +
  labs(title = "Upregulated in Poor Responders") +
  xlab("Hallmark Pathways") + 
  ylab("Normalized Enrichment Score (NES)") + 
  guides(fill = FALSE) +
  theme_prism(base_fontface = "plain")

ggarrange(suppfig3a, suppfig3b, ncol = 1, nrow = 2, heights = c(1, .3))
```

### Angiogenesis Gene Expression Plot

```{r angiogenesis-heatmap, fig.show='hide'}
blood_vessel <- scan("./input/blood_vessel_geneset.txt", character(), quote = "")
blood_vessel <- deseq %>% filter((gene %in% blood_vessel) &
  (enrichment != "none") &
  (abs(log2FoldChange) > 4))

goi <- blood_vessel$gene
b_v_heatmap <- counts %>% filter(gene %in% goi) %>% dplyr::select(-geneID)
rownames(b_v_heatmap) <- b_v_heatmap$gene
b_v_heatmap <- b_v_heatmap %>% dplyr::select(-gene)
b_v_heatmap <- as.data.frame(t(b_v_heatmap))
b_v_heatmap$sample <- rownames(b_v_heatmap)
rownames(b_v_heatmap) <- NULL
b_v_heatmap$group <- gsub("GBM[0-9]*_", "", b_v_heatmap$sample)
b_v_heatmap <- b_v_heatmap %>% pivot_longer(cols = -c("sample", "group"),
  names_to = "gene",
  values_to = "expression")

b_v_heatmap$log.expr <- log(b_v_heatmap$expression + 1)
b_v_heatmap$sample <- substr(b_v_heatmap$sample, 1, nchar(b_v_heatmap$sample) - 5)

fig1c <- ggplot(data = b_v_heatmap, mapping = aes(x = sample, y = gene, fill = log.expr)) +
  geom_tile() +
  scale_fill_gradient(high = "#ffb464", low = "#126079") +
  scale_colour_prism(palette = "colors") +
  #xlab(label = "Patient Derived Xenograft") + # Add a nicer x - axis title
  rremove("xlab") +
  facet_grid(~group,
    switch = "x", scales = "free_x", space = "free_x") +
  theme_prism() +
  theme(axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 60, vjust = 0.7),
    legend.position = "none")
fig1c
```

### Collagen Gene Expression Plot

```{r collagen-heatmap, fig.show='hide'}
collagen_genes_master <- c("COL1A1", "COL1A2", "COL2A1", "COL3A1", "COL4A1", "COL4A2", "COL4A3", "COL4A4", "COL4A5", "COL4A6", "COL5A1", "COL5A2", "COL5A3", "COL6A1", "COL6A2", "COL6A3", "COL6A4P1", "COL6A4P2", "COL6A5", "COL6A6", "COL7A1", "COL8A1", "COL8A2", "COL9A1", "COL9A2", "COL9A3", "COL10A1", "COL11A1", "COL11A2", "COL12A1", "COL13A1", "COL14A1", "COL15A1", "COL16A1", "COL17A1", "COL18A1", "COL19A1", "COL20A1", "COL21A1", "COL22A1", "COL23A1", "COL24A1", "COL25A1", "COL26A1", "COL27A1", "COL28A1")

collagen_genes <- deseq %>% filter(gene %in% collagen_genes_master) %>% filter(enrichment != "none")
collagen_genes <- collagen_genes$gene

collagen <- counts %>%
  filter(gene %in% collagen_genes) %>%
  dplyr::select(-geneID)
rownames(collagen) <- collagen$gene
collagen <- collagen %>%
  dplyr::select(-gene)
collagen <- as.data.frame(t(collagen))
collagen$sample <- rownames(collagen)
rownames(collagen) <- NULL
collagen$group <- gsub("GBM[0-9]*_", "", collagen$sample)
collagen <- collagen %>% pivot_longer(cols = -c("sample", "group"),
  names_to = "gene",
  values_to = "expression")

collagen$log.expr <- log(collagen$expression + 1)
collagen$sample <- substr(collagen$sample,
  1,
  nchar(collagen$sample) - 5)

fig1d <- ggplot(data = collagen, mapping = aes(x = sample, y = gene, fill = log.expr)) +
  geom_tile() +
  scale_fill_gradient(high = "#ffb464", low = "#126079") +
  scale_colour_prism(palette = "colors") +
  #xlab(label = "Patient Derived Xenograft") +
  rremove("xlab") +
  facet_grid(~group,
    switch = "x", scales = "free_x", space = "free_x") +
  theme_prism() +
  theme(axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 60, vjust = 0.7),
    legend.position = "bottom")
fig1d
```

### Figure 1

```{r fig1, fig.height=10, fig.width=15}
fig1 <- ggarrange(ggarrange(fig1a, NULL, fig1b, ncol = 1, nrow = 3, heights = c(1, 0.1, 1)), 
                fig1c, fig1d, ncol = 3, nrow = 1)
tiff("fig1.tiff", units="in", width=15, height=10, res=300)
fig1
dev.off()
```

# Unsupervised Clustering
```{r unsupervised DGE}
```

# Protein Level Analysis

## EGR1 Expression
```{r egr1-ihc-plot, fig.show='hide'}
egr1.poor.good <- read.csv("./input/ihc/egr1poorgood.csv") %>%
  mutate(bev_resp = recode(bev_resp, good = 'Good Response', poor = 'Poor Response'))

egr1.poor.good$fullname <- paste(egr1.poor.good$pdx_id, egr1.poor.good$animal_id)
fig2a <- ggplot(egr1.poor.good, aes(x = fct_rev(bev_resp), y = h_score)) +
  geom_boxplot(alpha = 0.5, fill = c("#ffb464", "#126079")) +
  geom_dotplot(aes(fill = factor(fullname)),
    binaxis = "y",
    stackdir = "center",
    dotsize = 0.5,
    binpositions = "all",
    stackgroups = TRUE) +
  theme_prism() +
  theme(
    legend.key.height = unit(10, "pt"),
    legend.title = element_blank(),
    legend.position = "none") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
  #guides(fill = guide_legend(nrow = 5))
  rremove("xlab") +
  ylab("Expression (H Score)") + 
  stat_compare_means(label.x = 2, label.y = NULL, aes(label = paste0("p = ", after_stat(p.format))))
fig2a
```

```{r eg1-poor-vs-placebo-ihc}
egr1.poor.placebo <- read.csv("./input/ihc/egr1poorplacebo.csv")
egr1.poor.placebo$fullname <- paste(egr1.poor.placebo$pdx_id, egr1.poor.placebo$animal_id)

poor.placebo.nuc <- ggplot(egr1.poor.placebo %>% dplyr::filter(staining_type == "nuclear"),
    aes(x = treatment, y = h_score)) +
  geom_boxplot(alpha = 0.5, fill = c("#ffb464", "#126079")) +
  geom_dotplot(aes(fill = factor(fullname)),
    binaxis = "y",
    stackdir = "center",
    dotsize = 0.5,
    binpositions = "all",
    stackgroups = TRUE) +
  theme_prism() +
  theme(
    legend.key.height = unit(10, "pt"),
    legend.title = element_text()) +
  guides(fill = guide_legend(title = "Animal ID")) +
  ggtitle("EGR1 Nuclear Expression") +
  xlab("Treatment Group") + ylab("Expression (H Score)")

poor.placebo.cyt <- ggplot(egr1.poor.placebo %>% dplyr::filter(staining_type == "cytoplasmic"),
    aes(x = treatment, y = h_score)) +
  geom_boxplot(alpha = 0.5, fill = c("#ffb464", "#126079")) +
  geom_dotplot(aes(fill = factor(fullname)),
    binaxis = "y",
    stackdir = "center",
    dotsize = 0.5,
    binpositions = "all",
    stackgroups = TRUE) +
  theme_prism() +
  theme(
    legend.key.height = unit(10, "pt"),
    legend.title = element_text()) +
  guides(fill = guide_legend(title = "Animal ID")) +
  #stat_compare_means(label.x.npc = "left", label.y.npc = "bottom") +
  ggtitle("EGR1 Cytoplasmic Expression") +
  xlab("Treatment Group") + ylab("Expression (H Score)")
exfig1 <- ggarrange(poor.placebo.nuc, poor.placebo.cyt, nrow = 1, ncol = 2)
```

```{r egr1-nuclear-expr-plot, fig.show='hide'}
egr1_nuclear_expr <- read_excel("./input/expr.xlsx", sheet = "egr1_nuclear") %>%
  mutate(bev_resp = recode(bev_resp, good_resp = 'Good Response', poor_resp = 'Poor Response'))

fig2c <- ggplot(egr1_nuclear_expr, aes(x = fct_rev(bev_resp), y = egr1_integrated_density)) +
  geom_boxplot(alpha = 0.5, fill = c("#ffb464", "#126079")) +
  geom_dotplot(aes(fill = factor(sample_id)),
    binaxis = "y",
    stackdir = "center",
    dotsize = 0.5,
    binpositions = "all",
    stackgroups = TRUE) +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
  theme_prism() +
  theme(
    legend.key.height = unit(10, "pt"),
    legend.title = element_blank(),
    legend.position = "none",) +
  #guides(fill = guide_legend(title = "Animal ID")) +
  rremove("xlab") +
  ylab("Nuclear Fluorescence Intensity") + 
  guides(fill = guide_legend(nrow = 5)) + 
  stat_compare_means(label.x = 2, label.y = NULL, aes(label = "p < 0.001"))
fig2c
```

### Figure 2

```{r fig2, fig.width=20, fig.height=6}
fig2 <- ggarrange(fig2a, NULL,  fig2c, NULL, nrow = 1, widths = c(1, 2, 1, 1))
tiff("fig2.tiff", units="in", width=15, height=10, res=300)
fig2
dev.off()
```

## Non-EGR1 Expression
### a7-nAChR Intensity

```{r a7-nAChR-expression-plot}
nachra7_expr <- read_excel("./input/expr.xlsx", sheet = "nachra7")
a7.nAChR.plot <- ggplot(nachra7_expr, aes(x = fct_rev(bev_resp), y = nachra7_integrated_density)) +
  geom_boxplot(alpha = 0.5, fill = c("#ffb464", "#126079")) +
  geom_dotplot(aes(fill = factor(sample_id)),
    binaxis = "y",
    stackdir = "center",
    dotsize = 0.5,
    binpositions = "all",
    stackgroups = TRUE) +
  theme_prism() +
  theme(legend.position = "none") +
  guides(fill = guide_legend(title = "Animal ID")) +
  xlab("Bevacizumab Response Group") + 
  ylab("Fluorescence Intensity (Integrated Density)")
```

### SOX10 Expression

```{r sox10-nuclear-expr}
sox10_nuclear_expr <- read_excel("./input/expr.xlsx", sheet = "sox10_nuclear") %>%
  mutate(bev_resp = recode(bev_resp, good_resp = 'Good Response', poor_resp = 'Poor Response'))
suppfig4a <- ggplot(sox10_nuclear_expr, aes(x = fct_rev(bev_resp), y = sox10_integrated_density_v2)) +
  geom_boxplot(alpha = 0.5, fill = c("#126079", "#ffb464")) +
  geom_dotplot(aes(fill = factor(sample_id)),
    binaxis = "y",
    stackdir = "center",
    dotsize = 0.5,
    binpositions = "all",
    stackgroups = TRUE) +
  theme_prism() +
  theme(legend.position = "none") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
  guides(fill = guide_legend(title = "Animal ID")) +
  rremove("xlab") +
  ylab("Nuclear Fluorescence Intensity")

suppfig4b <- ggplot(sox10_nuclear_expr, aes(x = fct_rev(bev_resp), y = number_of_sox10_cells)) +
  geom_boxplot(alpha = 0.5, fill = c("#126079", "#ffb464")) +
  geom_dotplot(aes(fill = factor(sample_id)),
    binaxis = "y",
    stackdir = "center",
    dotsize = 0.5,
    binpositions = "all",
    stackgroups = TRUE) +
  theme_prism() +
  theme(legend.position = "none") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
  guides(fill = guide_legend(title = "Animal ID")) +
  rremove("xlab") +
  ylab("Number of SOX10+ Cells")
```

```{r suppfig4, fig.height=6, fig.width=8}
suppfig4 <- ggarrange(suppfig4a, suppfig4b, nrow = 1, ncol = 2)
suppfig4
```

## Correlation between a7-nAChR and EGR1

```{r a7-nAChR-egr1-corr, fig.show='hide'}
nachra7_expr$field <- rep(c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10), 10)
combined_expr_table <- merge(x = egr1_nuclear_expr, y = nachra7_expr, by = c("sample_id", "field")) %>%
  select(c("sample_id", "field", "trt.x", "egr1_mean_intensity", "nachra7_mean_intensity")) #%>%
  #group_by(sample_id) %>% 
  #summarise_at(vars(-trt.x, -field), funs(mean(., na.rm=TRUE)))

fig3b <- ggplot(combined_expr_table, aes(x = egr1_mean_intensity, y = nachra7_mean_intensity)) +
  geom_point(aes(color = factor(sample_id))) +
  theme_prism() +
  theme(legend.position = "right") +
  guides(color = guide_legend(title = "Animal ID")) +
  xlab("EGR1 Intensity") + ylab("a7-nAChR Intensity") +
  stat_cor() + 
  geom_smooth(method=lm, se=FALSE, colour="black")
fig3b
```

```{r resid-plot}
egr1.nachr7.expr.corr <- lm(nachra7_mean_intensity ~ egr1_mean_intensity, data = combined_expr_table)
resid.plot <- ggplot(egr1.nachr7.expr.corr, aes(x = .fitted, y = .resid)) +
  geom_point() +
  geom_hline(yintercept = 0) +
  theme_prism() +
  ggtitle("e. Linear Model Residuals of EGR1 vs a7-nAChR") +
  xlab("Residuals") + ylab("Filtered")
```

### Figure 3
```{r fig3, fig.width=7, fig.height=10}
f3 <- ggarrange(NULL, NULL, fig3b, ncol = 1, nrow = 3, heights = c(1, 0.1, 1))
f3
```

# Survival Analysis
## GBM Analysis

```{r download-GBM-data}
gbm_cohort = XenaData %>%
  filter(XenaHostNames == "tcgaHub") %>%
  XenaScan("TCGA Glioblastoma")

cli = gbm_cohort %>%
  filter(DataSubtype == "phenotype") %>% # select clinical dataset
  XenaGenerate() %>% # generate a XenaHub object
  XenaQuery() %>%
  XenaDownload()
cli = XenaPrepare(cli)

ge = gbm_cohort %>%
  filter(DataSubtype == "protein expression RPPA", Label == "RPPA (replicate-base normalization)")

# download gene expression data
ge = gbm_cohort %>%
  filter(DataSubtype == "gene expression RNAseq", Label == "IlluminaHiSeq")
EGR1 = fetch_dense_values(host = ge$XenaHosts,
  dataset = ge$XenaDatasets,
  identifiers = "EGR1",
  use_probeMap = TRUE) %>% .[1, ]
EGR3 = fetch_dense_values(host = ge$XenaHosts,
  dataset = ge$XenaDatasets,
  identifiers = "EGR3",
  use_probeMap = TRUE) %>% .[1, ]
SOX10 = fetch_dense_values(host = ge$XenaHosts,
  dataset = ge$XenaDatasets,
  identifiers = "SOX10",
  use_probeMap = TRUE) %>% .[1, ]
RAMP3 = fetch_dense_values(host = ge$XenaHosts,
  dataset = ge$XenaDatasets,
  identifiers = "RAMP3",
  use_probeMap = TRUE) %>% .[1, ]
CHRNA7 = fetch_dense_values(host = ge$XenaHosts,
  dataset = ge$XenaDatasets,
  identifiers = "CHRNA7",
  use_probeMap = TRUE) %>% .[1, ]
```

### TCGA RNA Analysis
#### EGR1-Stratified Angiogenesis Expression

```{r tcga-egr1-angiogensis-rna-plot, fig.show='hide'}
EGR1_angiogensis = fetch_dense_values(host = ge$XenaHosts,
  dataset = ge$XenaDatasets,
  identifiers = c("EGR1", goi),
  use_probeMap = TRUE)
EGR1_angiogensis_matrix <- scale(t(EGR1_angiogensis))
EGR1_angiogensis_matrix <- EGR1_angiogensis_matrix[ , colSums(is.na(EGR1_angiogensis_matrix))==0]
EGR1_angiogensis <- as.data.frame(EGR1_angiogensis_matrix) %>% 
  mutate(group = ifelse(EGR1 > 1.5, "EGR1 upregulation", 
                        ifelse(EGR1 < -1.5, "EGR1 downregulation", "EGR1 normal"))) %>%
  rownames_to_column()
EGR1_angiogensis_pivoted <- pivot_longer(data = EGR1_angiogensis, 
                                         cols = -c(group, rowname),
                                         names_to = "Gene", 
                                         values_to = "Expression")

fig5a <- ggplot(data = EGR1_angiogensis_pivoted %>% 
                                  dplyr::filter(group != "EGR1 normal") %>%
                                  dplyr::filter(Gene != "EGR1"), 
                                mapping = aes(x = rowname, y = Gene, fill = Expression)) +
  geom_tile() +
  scale_fill_gradient(high = "#ffb464", low = "#126079") +
  scale_colour_prism(palette = "colors") +
  xlab(label = "TCGA Sample Number") + # Add a nicer x-axis title
  facet_grid(~group,
    switch = "x", scales = "free_x", space = "free_x") +
  theme_prism() +
  theme(axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        legend.position = "bottom")
fig5a
```

#### EGR1-Stratified Collagen Expression

```{r tcga-egr1-collagen-rna-plot, fig.show='hide'}
EGR_collagen = fetch_dense_values(host = ge$XenaHosts,
  dataset = ge$XenaDatasets,
  identifiers = c("EGR1", collagen_genes),
  use_probeMap = TRUE)
EGR_collagen_matrix <- scale(t(EGR_collagen))
EGR_collagen_matrix <- EGR_collagen_matrix[ , colSums(is.na(EGR_collagen_matrix))==0]
EGR_collagen <- as.data.frame(EGR_collagen_matrix) %>% 
  mutate(group = ifelse(EGR1 > 1.5, "EGR1 upregulation", 
                        ifelse(EGR1 < -1.5, "EGR1 downregulation", "EGR1 normal"))) %>%
  rownames_to_column()
EGR_collagen_pivoted <- pivot_longer(data = EGR_collagen, 
                                     cols = -c(group, rowname),
                                     names_to = "Gene", 
                                     values_to = "Expression")
fig5b <- ggplot(data = EGR_collagen_pivoted %>% 
                              dplyr::filter(group != "EGR1 normal") %>%
                              dplyr::filter(Gene != "EGR1"), 
                               mapping = aes(x = rowname, y = Gene, fill = Expression)) +
  geom_tile() +
  scale_fill_gradient(high = "#ffb464", low = "#126079") +
  scale_colour_prism(palette = "colors") +
  xlab(label = "TCGA Sample Number") + # Add a nicer x-axis title
  facet_grid(~group,
    switch = "x", scales = "free_x", space = "free_x") +
  theme_prism() +
  theme(axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        legend.position = "bottom")
fig5b
```

#### EGR1-Stratified Average Expression Boxplot

```{r tcga-angiogenesis-collagen-egr1-box-plot, fig.show='hide'}
blood_vessel <- scan("./input/blood_vessel_geneset.txt", character(), quote = "")
EGR1_collagen_angiogenesis = fetch_dense_values(host = ge$XenaHosts,
  dataset = ge$XenaDatasets,
  identifiers = unique(c("EGR1", collagen_genes_master, blood_vessel)),
  use_probeMap = TRUE)
EGR1_collagen_angiogenesis_matrix <- t(EGR1_collagen_angiogenesis)
EGR1_collagen_angiogenesis_matrix <- EGR1_collagen_angiogenesis_matrix[ , colSums(is.na(EGR1_collagen_angiogenesis_matrix))==0]
EGR1_collagen_angiogenesis <- as.data.frame(EGR1_collagen_angiogenesis_matrix) %>% 
  mutate(z = (EGR1 - mean(EGR1)) / sd(EGR1)) %>%
  mutate(EGR1 = case_when(
    z > 1.5~'high EGR1',
    z < -1.5~'low EGR1',
    (z < 1.5) & (z > -1.5) ~'EGR1 normal',
    TRUE~NA_character_
  )) %>%
  select(-c("z")) %>%
  rownames_to_column() 
EGR1_collagen_angiogenesis_pivoted <- pivot_longer(data = EGR1_collagen_angiogenesis, 
                                                   cols = -c(EGR1, rowname),
                                                   names_to = "Gene", 
                                                   values_to = "Expression") %>%
  dplyr::mutate(group = ifelse(Gene %in% goi, "Angiogenesis Genes", "Collagen Genes"))

my_comparisons <- list(c("Front", "Rear"),
                       c("Front", "4WD"),
                       c("Rear", "4WD"))

fig5c <- ggplot(EGR1_collagen_angiogenesis_pivoted %>% 
         dplyr::filter(EGR1 != "EGR1 normal"),
       aes(x = group, y = Expression, fill = EGR1)) +            # Applying ggplot function
  geom_boxplot(alpha = 0.5) +
  #stat_compare_means(aes(label = paste0("p = ", after_stat(p.format)))) +
  scale_fill_manual(values = c("#126079", "#ffb464")) + 
  theme_prism() +
  ylab("Log Expression") +
  rremove("xlab") +
  theme(legend.position = "right") +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10))

fig5c
```

#### EGR1-Stratified Survival Plot

```{r EGR1-survival, fig.width=10, fig.height=7, fig.show='hide'}
merged_EGR1 = tibble(sample = names(EGR1),
    EGR1_expression = as.numeric(EGR1)) %>%
  left_join(cli$GBM_survival.txt, by = "sample") %>%
  #filter(sample_type == "Primary Tumor") %>% # Keep only 'Primary Tumor'
  select(sample, EGR1_expression, OS.time, OS) %>%
  rename(time = OS.time,
    status = OS)

fit_EGR1 = coxph(Surv(time, status) ~EGR1_expression, data = merged_EGR1)
fit_EGR1

merged_EGR1 = merged_EGR1 %>%
  mutate(group = case_when(
    EGR1_expression > quantile(EGR1_expression, 0.9) ~'High',
    (EGR1_expression < quantile(EGR1_expression, 0.9) &
      EGR1_expression > quantile(EGR1_expression, 0.1)) ~'Normal',
    EGR1_expression < quantile(EGR1_expression, 0.1) ~'Low',
    TRUE~NA_character_
  )) %>%
  mutate(z = (EGR1_expression - mean(EGR1_expression)) / sd(EGR1_expression)) %>%
  mutate(group = case_when(
    z > 1.5~'High',
    z < -1.5~'Low',
    (z < 1.5) & (z > -1.5) ~'Normal',
    TRUE~NA_character_
  ))

fit_EGR1 = survfit(Surv(time, status) ~group,
  data = merged_EGR1)
fig5d <- ggsurvplot(fit_EGR1,
  pval = TRUE,
  pval.coord = c(600, 0.5),
  risk.table = TRUE,
  risk.table.height = 0.3,
  xlim = c(0, 1000),
  palette = c("#ffb464", "#126079", "grey"),
  legend.labs = c("High Expression", "Low Expression", "Normal Expression"),
  break.time.by = 250,
    ggtheme = theme_prism(),
    legend = c(0.8, 0.8))
fig5d
```

##### EGR1-Stratified Recurrent Tumor Survival Plot

```{r EGR1-recurrent-survival}
library("TCGAbiolinks")
library("limma")
library("edgeR")
library("glmnet")
library("factoextra")
library("FactoMineR")
library("caret")
library("SummarizedExperiment")
library("gplots")
library("survival")
library("survminer")
library("RColorBrewer")
library("gProfileR")
library("genefilter")

query_TCGA = GDCquery(
  project = "TCGA-GBM",
  data.category = "Transcriptome Profiling", # parameter enforced by GDCquery
  data.type = "Gene Expression Quantification", 
  experimental.strategy = "RNA-Seq",
  workflow.type = "STAR - Counts",
  sample.type = c("Recurrent Tumor", "Solid Tissue Normal"))

GDCdownload(query = query_TCGA)
tcga_data = GDCprepare(query_TCGA)

limma_pipeline = function(
  tcga_data,
  condition_variable,
  reference_group=NULL){

  design_factor = colData(tcga_data)[, condition_variable, drop=T]

  group = factor(design_factor)
  if(!is.null(reference_group)){group = relevel(group, ref=reference_group)}

  design = model.matrix(~ group)

  dge = DGEList(counts=assay(tcga_data),
                 samples=colData(tcga_data),
                 genes=as.data.frame(rowData(tcga_data)))

  # filtering
  keep = filterByExpr(dge,design)
  dge = dge[keep,,keep.lib.sizes=FALSE]
  rm(keep)

  # Normalization (TMM followed by voom)
  dge = calcNormFactors(dge)
  v = voom(dge, design, plot=TRUE)

  # Fit model to data given design
  fit = lmFit(v, design)
  fit = eBayes(fit)

  # Show top genes
  topGenes = topTable(fit, coef=ncol(design), number=Inf, sort.by="p")

  return(
    list(
      voomObj=v, # normalized data
      fit=fit, # fitted model and statistics
      topGenes=topGenes # the 100 most differentially expressed genes
    )
  )
}

limma_res = limma_pipeline(
  tcga_data=tcga_data,
  condition_variable="definition", # use standard
  #reference_group="Primary Tumor" # primary vs solid tissue 
)

d_mat = as.matrix(t(limma_res$voomObj$E))
expr_df = limma_res$topGenes

gene_id = "ENSG00000120738.8"
gene_name = "EGR1"

clinical = tcga_data@colData
clin_df = clinical[clinical$definition == "Recurrent Solid Tumor", 
                   c("patient", "vital_status", "days_to_death", "days_to_last_follow_up", "days_to_death", "gender", "prior_treatment")]

expr_diseased = d_mat[rownames(clin_df), gene_id]
expr_healthy = d_mat[setdiff(rownames(d_mat), rownames(clin_df)), gene_id]

clin_df$deceased = clin_df$vital_status == "Dead"
clin_df$overall_survival = ifelse(clin_df$deceased,
                                   clin_df$days_to_death,
                                   clin_df$days_to_last_follow_up)

clin_df$gene_value = d_mat[rownames(clin_df), gene_id]

# find the median value of the gene and print it
median_value = median(clin_df$gene_value)

clin_df$gene = ifelse(clin_df$gene_value >= median_value, "UP", "DOWN")

# we can fit a survival model, like we did in the previous section
fit = survfit(Surv(overall_survival, deceased) ~ gene, data=clin_df)
```

```{r suppfig7}
suppfig7 <- ggsurvplot(fit, data=clin_df, 
                       palette = c("#126079", "#ffb464"),
                       pval = TRUE,
                       risk.table = TRUE,
                       risk.table.col = "strata",
                       legend.labs = c("Low Expression", "High Expression"),
                       break.time.by = 250,
                       ggtheme = theme_prism(),
                       legend = c(0.7, 0.8))
suppfig7
```

### Figure 5

```{r fig5, fig.height=8, fig.width=12}
fig5alt <- ggarrange(ggarrange(fig5a, fig5b, ncol = 2, nrow = 1, common.legend = TRUE, legend = "none"), 
                NULL,
                ggarrange(fig5c, fig5d$plot, ncol = 2, nrow = 1, widths = c(0.5, 1)),
                ncol = 1, nrow = 3, heights = c(0.6, 0.1, 1))
fig5 <- ggarrange(ggarrange(fig5d$plot, fig5c, ncol = 2, nrow = 1, widths = c(1, 0.5)), 
                NULL,
                ggarrange(fig5a, fig5b, ncol = 2, nrow = 1, common.legend = TRUE, legend = "none"),
                ncol = 1, nrow = 3, heights = c(1, 0.1, 0.6))
tiff("fig5.tiff", units="in", width=15, height=10, res=300)
fig5
dev.off()
```

## Other Cancers
### Pan-Cancer Analysis

```{r pancan, eval=FALSE, fig.height=10, fig.width=10}
egr1.pancan.altered <- read_table("./input/survival/pancanEGR1_alt.txt")
egr1.pancan.altered$expression <- "altered"
egr1.pancan.unaltered <- read_table("./input/survival/pancanEGR1_unalt.txt")
egr1.pancan.unaltered$expression <- "unaltered"
egr1.pancan <- rbind(egr1.pancan.unaltered, egr1.pancan.altered)
egr1.pancan$Status = ifelse(egr1.pancan$Status == "censored", 1, 0)

fit_EGR1_pancan = survfit(Surv(Months, Status) ~expression, data = egr1.pancan)
EGR1_pancan_plot <- ggsurvplot(fit_EGR1_pancan,
  #xlim = c(0, 1000),
  palette = c("#ffb464", "#126079"),
  #conf.int = TRUE,
  #pval = TRUE,
  risk.table = TRUE,
  risk.table.col = "strata",
  legend.labs = c("Altered Expression", "Normal Expression"),
  ggtheme = theme_prism(),
  legend = c(0.7, 0.8),
  title = "EGR1-expression Stratisfied Survival Plot (Pan-Cancer Atlas)")
```

### Lung Cancer Analysis

```{r egr1-lung-survival, eval=FALSE}
lung_cohort = XenaData %>%
  filter(XenaHostNames == "tcgaHub") %>%
  XenaScan("TCGA Lung Cancer") 

cli_lung = lung_cohort %>%
  filter(DataSubtype == "phenotype") %>% # select clinical dataset
  XenaGenerate() %>% # generate a XenaHub object
  XenaQuery() %>%
  XenaDownload()
cli_lung = XenaPrepare(cli_lung)

lung_ge = lung_cohort %>%
  filter(DataSubtype == "gene expression RNAseq", Label == "IlluminaHiSeq")
lung_EGR1 = fetch_dense_values(host = lung_ge$XenaHosts,
  dataset = lung_ge$XenaDatasets,
  identifiers = "EGR1",
  use_probeMap = TRUE) %>% .[1, ]

merged_lung_EGR1 = tibble(sample = names(lung_EGR1),
    EGR1_expression = as.numeric(lung_EGR1)) %>%
  left_join(cli_lung$LUNG_survival.txt, by = "sample") %>%
  #filter(sample_type == "Primary Tumor") %>% # Keep only 'Primary Tumor'
  select(sample, EGR1_expression, OS.time, OS) %>%
  rename(time = OS.time,
    status = OS) %>%
  mutate(group = case_when(
    EGR1_expression > quantile(EGR1_expression, 0.9) ~'High',
    (EGR1_expression < quantile(EGR1_expression, 0.9) &
      EGR1_expression > quantile(EGR1_expression, 0.1)) ~'Normal',
    EGR1_expression < quantile(EGR1_expression, 0.1) ~'Low',
    TRUE~NA_character_
  )) %>%
  mutate(z = (EGR1_expression - mean(EGR1_expression)) / sd(EGR1_expression)) %>%
  mutate(group = case_when(
    z > 1.5~'High',
    z < -1.5~'Low',
    (z < 1.5) & (z > -1.5) ~'Normal',
    TRUE~NA_character_
  ))

fit_EGR1_lung = survfit(Surv(time, status) ~group,
  data = merged_lung_EGR1 %>% dplyr::filter(group != "Low"))

EGR1_lung_plot <- ggsurvplot(fit_EGR1_lung,
  pval = TRUE,
  pval.coord = c(600, 0.5),
  #xlim = c(0, 1000),
  palette = c("#ffb464", "#126079"),
  #conf.int = TRUE,
  #pval = TRUE,
  risk.table = TRUE,
  risk.table.col = "strata",
  legend.labs = c("High Expression", "Normal Expression"),
  surv.median.line = "hv",
  break.time.by = 250,
    ggtheme = theme_prism(),
    legend = c(0.7, 0.8),
    title = "EGR1-expression Stratisfied Survival Plot (nSCLC)")
```

## Other Genes
### CHRNA7 Survival Plot

```{r CHRNA7-survival}
merged_CHRNA7 = tibble(sample = names(CHRNA7),
    CHRNA7_expression = as.numeric(CHRNA7)) %>%
  left_join(cli$GBM_survival.txt, by = "sample") %>%
  #filter(sample_type == "Primary Tumor") %>% # Keep only 'Primary Tumor'
select(sample, CHRNA7_expression, OS.time, OS) %>%
  rename(time = OS.time,
    status = OS)

merged_CHRNA7 = merged_CHRNA7 %>%
  mutate(group = case_when(
    CHRNA7_expression > quantile(CHRNA7_expression, 0.9) ~ 'High',
    (CHRNA7_expression < quantile(CHRNA7_expression, 0.9) &
      CHRNA7_expression > quantile(CHRNA7_expression, 0.1)) ~ 'Normal',
    CHRNA7_expression < quantile(CHRNA7_expression, 0.1) ~ 'Low',
    TRUE ~ NA_character_
  )) %>%
  mutate(z = (CHRNA7_expression - mean(CHRNA7_expression)) / sd(CHRNA7_expression)) %>%
  mutate(group = case_when(
    z > 1.5~'High',
    z < -1.5~'Low',
    (z < 1.5) & (z > -1.5) ~'Normal',
    TRUE~NA_character_
  ))

fit_CHRNA7 = survfit(Surv(time, status) ~group,
  data = merged_CHRNA7 %>% dplyr::filter(group != "Low"))
suppfig8a <- ggsurvplot(fit_CHRNA7,
  pval = TRUE,
  pval.coord = c(600, 0.5),
  #xlim = c(0, 1000),
  palette = c("#ffb464", "#126079"),
  #conf.int = TRUE,
  #pval = TRUE,
  risk.table = TRUE,
  risk.table.col = "strata",
  legend.labs = c("High Expression", "Normal Expression"),
  surv.median.line = "hv",
  break.time.by = 250,
    ggtheme = theme_prism(),
    legend = c(0.7, 0.8),
    title = "CHRNA7-Stratisfied Survival Plot")
```

### SOX10 Survival Plot

```{r SOX10-survival}
#SOX10-- --
merged_SOX10 = tibble(sample = names(SOX10),
    SOX10_expression = as.numeric(SOX10)) %>%
  left_join(cli$GBM_survival.txt, by = "sample") %>%
  select(sample, SOX10_expression, OS.time, OS) %>%
  rename(time = OS.time,
    status = OS)
#ggplot(merged_SOX10, aes(x = SOX10_expression)) + geom_histogram(color="black", fill="white") + theme_prism()

merged_SOX10 = merged_SOX10 %>%
  mutate(group = case_when(
    SOX10_expression > quantile(SOX10_expression, 0.9) ~'High',
    (SOX10_expression < quantile(SOX10_expression, 0.9) &
      SOX10_expression > quantile(SOX10_expression, 0.1)) ~'Normal',
    SOX10_expression < quantile(SOX10_expression, 0.1) ~'Low',
    TRUE~NA_character_
  )) %>%
  mutate(z = (SOX10_expression - mean(SOX10_expression)) / sd(SOX10_expression)) %>%
  mutate(group = case_when(
    z > 1.5~'High',
    z < -1.5~'Low',
    (z < 1.5) & (z > -1.5) ~'Normal',
    TRUE~NA_character_
  ))
fit_SOX10 = survfit(Surv(time, status) ~group,
  data = merged_SOX10 %>% dplyr::filter(group != "Low"))
suppfig8b <- ggsurvplot(fit_SOX10,
  pval = TRUE,
  pval.coord = c(600, 0.5),
  #xlim = c(0, 1000),
  palette = c("#ffb464", "#126079"),
  #conf.int = TRUE,
  #pval = TRUE,
  risk.table = TRUE,
  risk.table.col = "strata",
  legend.labs = c("High Expression", "Normal Expression"),
  surv.median.line = "hv",
  break.time.by = 250,
    ggtheme = theme_prism(),
    legend = c(0.7, 0.8),
    title = "SOX10-Stratisfied Survival Plot")
```

### EGR3 Survival Plot

```{r EGR3-survival, eval=FALSE}
#EGR3-- --
merged_EGR3 = tibble(sample = names(EGR3),
    EGR3_expression = as.numeric(EGR3)) %>%
  left_join(cli$GBM_survival.txt, by = "sample") %>%
  #filter(sample_type == "Primary Tumor") %>% # Keep only 'Primary Tumor'
  select(sample, EGR3_expression, OS.time, OS) %>%
  rename(time = OS.time,
    status = OS)

fit_EGR3 = coxph(Surv(time, status) ~EGR3_expression, data = merged_EGR3)
fit_EGR3

merged_EGR3 = merged_EGR3 %>%
  mutate(group = case_when(
    EGR3_expression > quantile(EGR3_expression, 0.9) ~'High',
    (EGR3_expression < quantile(EGR3_expression, 0.9) &
      EGR3_expression > quantile(EGR3_expression, 0.1)) ~'Normal',
    EGR3_expression < quantile(EGR3_expression, 0.1) ~'Low',
    TRUE~NA_character_
  )) %>%
  mutate(z = (EGR3_expression - mean(EGR3_expression)) / sd(EGR3_expression)) %>%
  mutate(group = case_when(
    z > 1.5~'High',
    z < -1.5~'Low',
    (z < 1.5) & (z > -1.5) ~'Normal',
    TRUE~NA_character_
  ))
fit_EGR3 = survfit(Surv(time, status) ~group,
  data = merged_EGR3 %>% dplyr::filter(group != "Low"))
suppfig8c <- ggsurvplot(fit_EGR3,
  pval = TRUE,
  pval.coord = c(600, 0.5),
  #xlim = c(0, 1000),
  palette = c("#ffb464", "#126079"),
  #conf.int = TRUE,
  #pval = TRUE,
  risk.table = TRUE,
  risk.table.col = "strata",
  legend.labs = c("High Expression", "Normal Expression"),
  surv.median.line = "hv",
  break.time.by = 250,
    ggtheme = theme_prism(),
    legend = c(0.7, 0.8))
```

### RAMP3 Survival Plot

```{r RAMP3-survival, eval=FALSE}
merged_RAMP3 = tibble(sample = names(RAMP3),
    RAMP3_expression = as.numeric(RAMP3)) %>%
  left_join(cli$GBM_survival.txt, by = "sample") %>%
  #filter(sample_type == "Primary Tumor") %>% # Keep only 'Primary Tumor'
select(sample, RAMP3_expression, OS.time, OS) %>%
  rename(time = OS.time,
    status = OS)

fit = coxph(Surv(time, status) ~RAMP3_expression, data = merged_RAMP3)
fit

merged_RAMP3 = merged_RAMP3 %>%
  mutate(group = case_when(
    RAMP3_expression > quantile(RAMP3_expression, 0.9) ~'High',
    (RAMP3_expression < quantile(RAMP3_expression, 0.9) &
      RAMP3_expression > quantile(RAMP3_expression, 0.1)) ~'Normal',
    RAMP3_expression < quantile(RAMP3_expression, 0.1) ~'Low',
    TRUE~NA_character_
  )) %>%
  mutate(z = (RAMP3_expression - mean(RAMP3_expression)) / sd(RAMP3_expression)) %>%
  mutate(group = case_when(
    z > 1.5~'High',
    z < -1.5~'Low',
    (z < 1.5) & (z > -1.5) ~'Normal',
    TRUE~NA_character_
  ))
fit_RAMP3 = survfit(Surv(time, status) ~group,
  data = merged_RAMP3 %>% dplyr::filter(group != "Low"))
suppfig8d <- ggsurvplot(fit_RAMP3,
  pval = TRUE,
  pval.coord = c(600, 0.5),
  #xlim = c(0, 1000),
  palette = c("#ffb464", "#126079"),
  #conf.int = TRUE,
  #pval = TRUE,
  risk.table = TRUE,
  risk.table.col = "strata",
  legend.labs = c("High Expression", "Normal Expression"),
  surv.median.line = "hv",
  break.time.by = 250,
    ggtheme = theme_prism(),
    legend = c(0.7, 0.8))
```

### Other Genes Plot

```{r suppfig8, fig.height=10, fig.width=7}
splots <- list()
splots[[1]] <- suppfig8a
splots[[2]] <- suppfig8b

suppfig8 <- arrange_ggsurvplots(splots, print = FALSE, ncol = 1, nrow = 2, risk.table.height = 0.35)
suppfig8
```